src/explore-education-statistics-admin/src/pages/release/data/components/DataFileReplacementPlan.tsx

```diff
@@ -73,6 +79,54 @@ const DataFileReplacementPlan = ({
     () => plan?.footnotes.some(footnote => !footnote.valid) ?? false,
     [plan],
   );
+  const isNewReplaceDsvFeatureEnabled = useFeatureFlag(
+    'enableReplacementOfPublicApiDataSets',
+  );
+
+  const hasDataSetVersionPlan = useMemo<boolean>(
+    () =>
+      isNewReplaceDsvFeatureEnabled &&
+      plan?.apiDataSetVersionPlan !== undefined,
+    [plan, isNewReplaceDsvFeatureEnabled],
+  );
+
+  const hasInvalidLocationMapping = useMemo<boolean>(
+    () =>
+      hasDataSetVersionPlan
+        ? !(
+            plan?.apiDataSetVersionPlan?.mappingStatus?.locationsComplete ??
+            false
+          )
+        : false,
+    [plan, hasDataSetVersionPlan],
+  );
+
+  const hasInvalidFilterMapping = useMemo<boolean>(
+    () =>
+      hasDataSetVersionPlan
+        ? !(
+            plan?.apiDataSetVersionPlan?.mappingStatus?.filtersComplete ?? false
+          )
+        : false,
+    [plan, hasDataSetVersionPlan],
+  );
```

Comment:
feel free to disregard this one, I felt the memo's could be consolidated into one

```tsx
const {
  hasDataSetVersionPlan,
  hasInvalidLocationMapping,
  hasInvalidFilterMapping,
} = useMemo(()=>{
  if(!isNewReplaceDsvFeatureEnabled) {
    return {
      hasDataSetVersionPlan: false,
      hasInvalidLocationMapping: false,
      hasInvalidFilterMapping: false,
    }
  }

  return {
    hasDataSetVersionPlan: !!plan?.apiDataSetVersionPlan,
    hasInvalidLocationMapping:  !!plan?.apiDataSetVersionPlan?.mappingStatus?.locationsComplete,
    hasInvalidFilterMapping: !!plan?.apiDataSetVersionPlan?.mappingStatus?.filtersComplete,
  }

  
}, [plan, isNewReplaceDsvFeatureEnabled ])
```